cmake/CMakeDetermineMetalCompiler.cmake

```diff
@@ -0,0 +1,83 @@
+# Distributed under the OSI-approved BSD 3-Clause License.  See accompanying
+# file LICENCE.txt or https://cmake.org/licensing for details.
+
+# CMakeDetermine(LANG)Compiler.cmake -> this should find the compiler for LANG and configure CMake(LANG)Compiler.cmake.in
+
+include(${CMAKE_ROOT}/Modules/CMakeDetermineCompiler.cmake)
+
+if(NOT CMAKE_Metal_COMPILER_NAMES)
+    set(CMAKE_Metal_COMPILER_NAMES metal)
+endif()
+
+if("${CMAKE_GENERATOR}" STREQUAL "Xcode")
+    set(CMAKE_Metal_COMPILER_XCODE_TYPE sourcecode.metal)
+
+    execute_process(COMMAND xcrun --find metal
+        OUTPUT_VARIABLE _xcrun_out OUTPUT_STRIP_TRAILING_WHITESPACE
+        ERROR_VARIABLE _xcrun_err RESULT_VARIABLE _xcrun_result)
+    if(_xcrun_result EQUAL 0 AND EXISTS "${_xcrun_out}")
+        set(CMAKE_Metal_COMPILER "${_xcrun_out}")
+    else()
+        _cmake_find_compiler_path(Metal)
+    endif()
+else()
+    if(CMAKE_Metal_COMPILER)
+        _cmake_find_compiler_path(Metal)
+    else()
+        set(CMAKE_Metal_COMPILER_INIT NOTFOUND)
+
+        if(NOT $ENV{METALC} STREQUAL "")
+            get_filename_component(CMAKE_Metal_COMPILER_INIT $ENV{METALC} PROGRAM PROGRAM_ARGS CMAKE_Metal_FLAGS_ENV_INIT)
+            if(CMAKE_Metal_FLAGS_ENV_INIT)
+                set(CMAKE_Metal_COMPILER_ARG1 "${CMAKE_Metal_FLAGS_ENV_INIT}" CACHE STRING "Arguments to the Metal compiler")
+            endif()
+            if(NOT EXISTS ${CMAKE_Metal_COMPILER_INIT})
```

Comment:
I wonder if `${CMAKE_Metal_COMPILER_INIT}` should be quoted?

Reply:
None of the other language support files built-in to CMake seem to quote the `CMAKE_whatever_COMPILER_INIT` stuff when they test it 🤷🏼‍♂️ 

It should be a filepath to a compiler executable, as I understand it

Reply:
The reason it should *probably* be quoted is without the quotes, CMake *generally* just inserts the contents of the variable directly into the if command. This means if you have a semicolon, CMake starts thinking about lists, and if you have a space, you've begun a new argument to the if command. If the variable is quoted, then none of that applies, and CMake will just test if the path exists.

Welcome to the world of CMake's idiosyncrasies.